app/models/concerns/user_omniauth_concern.rb

```diff
@@ -0,0 +1,52 @@
+module UserOmniauthConcern
```

Comment:
Only `#new_with_session` need to consider different providers, check my comments there ?